scripts/genTypes.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs')
```

Comment:
this logic should be tested, there are a lot of edge cases so it's probably not fully working as is, tests will make it possible to fix it

node protodef has test values